CONTRIBUTING.md

```diff
@@ -6,11 +6,77 @@ _As contributors and maintainers of this project, and in the interest of fosteri
 
 ## Getting Started
 
-We have full documentation on how to get started contributing here:
+### Repo layout
+
+```
+├── publishers # Home for all functions metadata
+│   ├── kustomize
+│   │   ├── fn-foo.yaml
+│   │   ├── fn-bar.yaml
+│   │   ├── README.md
+│   │   └── OWNERS # OWNERS of the publisher
+│   ├── kubeflow
+│   ├── sig-cli
+│   └── OWNERS # OWNERS to approve new publishers
+├── krm-functions # Home for in-tree functions source code
+│   ├── Makefile
+│   ├── kustomize
+│   │   ├── fn-foo
+│   │   ├── README.md
+│   │   └── OWNERS # OWNERS to approve code change to the function
+│   └── sig-cli
+├── site 
+├── Makefile
+└── OWNERS
+```
+
+## Contributing in-tree KRM function source code
+
+This section discusses how to contribute KRM function source code, so that it can be managed and released by this repo.
+
+For each in-tree function, the implementation is in the `krm-functions/` directory. Each publisher will need to create their own subdirectory `krm-functions/{PUBLISHER}/` 
+in this directory to store their functions source code. For example, SIG-CLI sponsored functions are located
+under `krm-functions/sig-cli/`.
+
+Each function must be in its own subdirectory `krm-functions/{PUBLISHER}/{FUNCTION NAME}/`. This directory should 
+contain: 
```

Comment:
Some of the content here is duplicate of `krm-functions/README.md`. I guess we can point to that doc here.